src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java

```diff
@@ -194,13 +243,45 @@ public void should_retrieve_user_multiple_profiles_without_roles_when_idam_fails
         assertThat(profile.getEmail()).isEqualTo(up.getEmail());
         assertThat(profile.getFirstName()).isEqualTo(up.getFirstName());
         assertThat(profile.getLastName()).isEqualTo(up.getLastName());
-        assertThat(profile.getRoles().size()).isEqualTo(0);
+        assertThat(profile.getRoles().size()).isZero();
         assertThat(profile.getErrorMessage()).isNotEmpty();
         assertThat(profile.getErrorStatusCode()).isEqualTo("404");
 
         Mockito.verify(auditRepository, Mockito.times(1)).save(any(Audit.class));
     }
 
+    @Test
+    public void should_retrieve_user_multiple_profiles_with_roles_when_idam_success() {
+        idamRolesInfo = new IdamRolesInfo(entity, HttpStatus.OK);
+
+        UserProfile up = UserProfileTestDataBuilder.buildUserProfile();
+        up.setStatus(IdamStatus.ACTIVE);
+        UserProfile upMock = mock(UserProfile.class);
+        when(upMock.getIdamId()).thenReturn(up.getIdamId());
+        when(upMock.getEmail()).thenReturn(up.getEmail());
+        when(upMock.getFirstName()).thenReturn(up.getFirstName());
+        when(upMock.getLastName()).thenReturn(up.getLastName());
+        when(upMock.getStatus()).thenReturn(up.getStatus());
+
+        when(idamServiceMock.fetchUserById(any(String.class))).thenReturn(idamRolesInfo);
+        when(auditRepository.save(any())).thenReturn(audit);
+
+        UserProfile profile = userProfileRetriever.getRolesFromIdam(upMock, true);
+
+        assertThat(profile).isNotNull();
+        assertThat(profile.getEmail()).isEqualTo(up.getEmail());
+        assertThat(profile.getFirstName()).isEqualTo(up.getFirstName());
+        assertThat(profile.getLastName()).isEqualTo(up.getLastName());
+        assertThat(profile.getRoles().size()).isZero();
+        assertThat(profile.getErrorMessage()).isNull();
+        assertThat(profile.getErrorStatusCode()).isNull();
+
+        verify(auditRepository, Mockito.times(1)).save(any(Audit.class));
+        verify(upMock, times(1)).setRoles(any(IdamRolesInfo.class));
+        verify(upMock, times(1)).setErrorMessage(any(String.class));
+        verify(upMock, times(1)).setErrorStatusCode(any(String.class));
+    }
+
     @Test
     public void should_throw_404_single_user_profile_without_roles_when_idam_fails() {
         idamRolesInfo = new IdamRolesInfo(entity, HttpStatus.NOT_FOUND);
```

Comment:
prefix test for method

Reply:
This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501